### PR TITLE
Builtins refactoring

### PR DESCRIPTION
**Pull request by folex**

### CODE REVIEW

File: particle-closures/src/host_closures.rs [hunk 1]
```diff
@@ -200,6 +212,7 @@ impl<C: Clone + Send + Sync + 'static + AsRef<KademliaApi> + AsRef<ConnectionPoo
     }
 
     fn remove_script(&self, args: Args, params: ParticleParameters) -> Result<JValue, JError> {
+        println!("remove_script args: {:#?}", args);
```
Comment (DieMyst) [inline particle-closures/src/host_closures.rs:215]: println

File: particle-modules/src/modules.rs [hunk 2]
```diff
@@ -180,7 +178,28 @@ impl ModuleRepository {
...
+                let hash = ModuleHash::from_hex(&hash)?;
+                let path = modules_dir.join(hash.config_file_name());
+                let interface =
+                    module_interface(&path).wrap_err(f!("parse interface ${path:?}"))?;
+
+                json!(interface)
+            };
+            interface.map_err(|err| {
+                json!(format!("{:?}", err)
+                    .split("Stack backtrace:")
```
Comment (DieMyst) [inline particle-modules/src/modules.rs:199]: leave a comment here about what's going on

File: particle-modules/src/modules.rs [hunk 3]
```diff
@@ -180,7 +178,28 @@ impl ModuleRepository {
...
+                let hash: String = Args::next("hash", &mut args)?;
+                let hash = ModuleHash::from_hex(&hash)?;
+                let path = modules_dir.join(hash.config_file_name());
+                let interface =
+                    module_interface(&path).wrap_err(f!("parse interface ${path:?}"))?;
+
+                json!(interface)
+            };
+            interface.map_err(|err| {
+                json!(format!("{:?}", err)
```
Comment (DieMyst) [inline particle-modules/src/modules.rs:198]: why json as an error, but not a struct?

**Review by DieMyst — APPROVED** [review]

File: particle-modules/src/modules.rs (hunk 3 above)
Comment (folex) [reply particle-modules/src/modules.rs:198]: it's `Result<JValue, JValue>` as required by `closure` function

**Pull request merged**